### ISSUE TITLE
Bug: Self-signed LND certificates error in Windows Desktop

### ISSUE THREAD

**Issue by rolznz** [getAlby/hub#405]
From https://primal.net/e/note16l86pvcfcqdkrpm7h0re58gn9420js2tszndxqh08xfgvqe2lxpqph82r0

> (1) There seems to be an issue when LND uses self-signed certificates, which is the default behavior. AlbyHub complains that the certificate isn't trusted, and this happens even if the self-signed certificate is added to the Trusted Root Authorities folder in the Windows Certs console. What I ended up doing was creating a dedicated DNS entry for LND (lnd.mydomain.com), and then getting a free SSL cert from one of the countless free providers, including that DNS name on the certificate. I forced LND to use that certificate instead of the self-signed one and then referenced it in the AlbyHub setup. That seemed to clear up the error.

**Comment by rolznz** [comment]
I wonder why there is no issue when using Polar (at least on Linux). What happens if you provide no certificate? in the `lnclient/lnd/wrapper/lnd.go` we could try `creds = credentials.NewTLS(&tls.Config{InsecureSkipVerify: true})`

@im-adithya are you able to reproduce the issue?

**Comment by Filouman** [comment]
Adding my specs to the list. Self-hosted on x86 linux machine.

`tlsencryptkey=false` is set in lnd.conf and have regenerated certs, however the issue persists.

This definitely appears to be an AlbyHub specific issue, since all other apps are able to accept the self-signed certs that LND generates without any issue.

**Comment by Filouman** [comment]
Issue persists on AlbyHub v1.11.2

**Comment by rolznz** [comment]
We didn't implement any fix yet. @Filouman do you have any coding experience? are you able to test the suggested fix above? https://github.com/getAlby/hub/issues/405#issuecomment-2504719186

**Comment by Filouman** [comment]
No coding experience, however I was able to try the suggested fix and recompile– That did not solve the issue, at least on my system.

**Comment by rolznz** [comment]
@Filouman did you try it and pass an empty certificate?

**Comment by Filouman** [comment]
> @Filouman did you try it and pass an empty certificate?

I'm not sure how I could do that? If I try to leave TLS Certificate (Hex) field blank I get 
<img width="305" alt="Screenshot 2024-12-10 at 12 31 42 PM" src="https://github.com/user-attachments/assets/ea544d6f-32b0-417c-8225-048a57e13e9d">

```
{"error":"invalid character '\u003c' looking for beginning of value","level":"error","msg":"Failed to decode API response","time":"2024-12-10T17:29:42Z"}
{"error":"invalid character '\u003c' looking for beginning of value","level":"error","msg":"Failed to request alby info endpoint","time":"2024-12-10T17:29:42Z"}
```

**Comment by im-adithya** [comment]
Can you please try if the fix works?

You can find the executables to test here:
Desktop: https://github.com/getAlby/hub/actions/runs/12277974396?pr=877
Server: https://github.com/getAlby/hub/actions/runs/12277974400?pr=877

**Comment by Filouman** [comment]
Yes, that works for me if I leave TLS Certificate (Hex) field blank during setup.

**Comment by Filouman** [comment]
Interestingly, I am getting a different error on v.1.12.0
<img width="295" alt="Screenshot 2024-12-19 at 12 26 42 PM" src="https://github.com/user-attachments/assets/f61b768e-784a-44fa-a437-d0e12e7cc573" />
Same error with automatic setup or advance setup and both when entering a certificate HEX and leaving cert field blank.

Is there any difference between v.1.12.0 and the executables posted by @im-adithya above? https://github.com/getAlby/hub/issues/405#issuecomment-2536083959

**Comment by im-adithya** [comment]
> Interestingly, I am getting a different error on v.1.12.0

That error is shown when `cp.AppendCertsFromPEM(cert)` part fails, not sure why exactly that happens but is it possible that your certificates are not PEM encoded? You can check using this: [Stack Overflow link](https://stackoverflow.com/questions/29924053/validating-that-an-uploaded-file-is-a-valid-pem-file)

> Is there any difference between v.1.12.0 and the executables posted above?

Yes, there is a small difference that we verify certficates when they are provided by the user (in the builds we skipped verification even if provided by user), apart from that the scenario where no certificate is passed is same (we thought if passing certificate fails then they can choose to proceed without it)

> Same error with automatic setup or advance setup and both when entering a certificate HEX and leaving cert field blank.

The error even after shifting to blank certificate might be because the certificate added to the db config before wasn't removed, did you try doing it freshly without a certificate? But this shouldn't happen ideally, will make a fix to always consider the new configuration instead of using a previously passed certificate.
